bounded_sum_known_n/BoundedSumKnownN.tex

```diff
@@ -0,0 +1,394 @@
+    \[
+        d_{Abs}(\function(v), \function(w)) \leq d_{Sym}(v, w) \cdot \dfrac{\texttt{U-L}}{2}.
+    \]
+    This would imply that
+    \begin{equation}\label{eq:abs1}
+        d_{Abs}(\function(v), \function(w)) \leq d_{Sym}(v, w) \cdot \dfrac{\texttt{U-L}}{2} \leq \din \cdot \dfrac{\texttt{U-L}}{2},
+    \end{equation}
+    and by the stability relation this will imply that
+    \begin{equation}\label{eq:abs2}
+        d_{Abs}(\function(v), \function(w)) \leq \dout,
+    \end{equation}
+    as we want to see. 
+\end{proof}
+
+\subsection{First proof: using the path property (adjacent pairs approach)}
+
+To show that $d_{Abs}(\function(v), \function(w)) \leq d_{Sym}(v, w) \cdot \frac{\texttt{U-L}}{2}$, we will use the three lemmas described in the section ``The path property of symmetric distance on sized domains" from Section 5.2 in the document \href{https://www.overleaf.com/project/60d214e390b337703d200982}{``List of definitions used in the proofs"}. With these three lemmas, which are applicable to \texttt{MakeBoundedSumN} because \texttt{input\_domain} is a a sized domain and \texttt{input\_metric} is symmetric distance, it suffices to show the following: For all vectors $x, y \in$ \texttt{input\_domain} such that $d_{Sym}(x, y) = 2$, it follows that 
+\[
+d_{Abs}(\texttt{function}(x), \texttt{function}(y)) \leq \texttt{U} - \texttt{L}.
+\]
+By Lemma 5.6 from \href{https://www.overleaf.com/project/60d214e390b337703d200982}{``List of definitions used in the proofs"}, we know that vectors $x, y$ only differ on one element, given that, by assumption, $d_{Sym}(x, y) = 2$. Wlog, let this different element be the $k$-th element of $x$ and $y$, where $x_k = \alpha$, $y_k = \beta$ with $\alpha \neq \beta$.\footnote{The first element of a vector is indexed by 0.} Then,
+\[
+    d_{Abs}(\texttt{function}(x), \texttt{function}(y)) = |\texttt{function}(x) - \texttt{function}(y)| = 
+\]
+\[
+    = \Big|\sum_{i=0}^{\texttt{n}-1}x_i - \sum_{i=0}^{\texttt{n}-1}y_i\Big| = \Big| \sum_{i=0}^{\texttt{n}-1} (x_i - y_i) \Big| = |0 + (x_k - y_k)| = |\alpha - \beta| \leq |\texttt{U-L}| = \texttt{U-L},
+\]
+since \texttt{U} $\geq$ \texttt{L}. Therefore, applying Lemma 5.7 from \href{https://www.overleaf.com/project/60d214e390b337703d200982}{``List of definitions used in the proofs"}, it follows that \texttt{function} is $($\texttt{U-L}$)/2$-stable. By definition, this implies that for any $v, w \in$ \texttt{input\_domain},
+\[
+    d_{Abs}(\texttt{function}(v), \texttt{function}(w)) \leq d_{Sym}(v, w) \cdot (\texttt{U-L})/2.
+\]
+Lastly, by Equations~\ref{eq:abs1} and \ref{eq:abs2} this implies that
+\[
+    d_{Abs}(\function(v), \function(w)) \leq \dout,
+\]
+as we want to prove.
+
+%\silvia{Flag: this will be updated to the more general notion of path property (through shortest path metric on a graph), but this matches the current version of the proofs document.}
+
+\newpage
+
+\subsection{Second proof: direct method (all pairs approach)}
```

Comment:
I would drop the direct proof given that the one using the path property is much more readable.

Reply:
Removed. We can readily retrieve it if necessary by (1) looking in the Overleaf history or (2) looking at the github history.